bfh/fields.py

```diff
@@ -63,7 +64,10 @@ def __get__(self, instance, cls=None):
         return instance.__dict__.get(self.field_name)
 
     def __set__(self, instance, value):
-        instance.__dict__[self.field_name] = value
+        if value is None:
```

Comment:
I sort of agree, sort of disagree. I agree that this library should store the default as a reference so that if someone wants to shoot self in foot in the very pythonic manner described, they are free to do so. Disagree that we can have all the desired behavior if the implementation only touches the setter.

N.B. the example I provided was not a primary use-case, just an illustration of the danger of using a shared mutable value in a descriptor class like this. Even if it's not a usecase though, this object should treat state in a predictable and pythonic manner.

Let's talk interface though. Ergonomics. I think that explicit None passed to setter causing default to be set is bizarre. As a user, i'm expecting it will show up when the field is not set at all -- this is the meaning of the word 'default'.

Here are the behaviors I think seem sane:

a. If a schema obj is instantiated without assigning a specific value to a field, the default should show up.

```python
> wow = Wow()
> wow.foo
# {"wow": 1}
```

b. A field can be set to something else, then back to `None`, and the default should show up.

```python
> Wow(foo={"zip": "pow"})
> wow.foo
# {"zip": "pow"}
> Wow(foo=None)
> wow.foo
# {"wow": 1}
```

c. You can mutate the default value (but this shouldn't affect other instances)
```python
> wow2 = Wow()
> wow.foo['wow'] = 5
> wow.foo
# {"wow": 5}
> wow2.foo
# {"wow": 1}
```

d. Combining (b) and (c), there should be no 'memory' if the value is set to something else and then back to None. I.e. setting value to None gives a new reference to a fresh default value.
```python
> wow = Wow()
> wow.foo
# {"wow": 1}
> wow.foo['wow'] = 5
> wow.foo
# {"wow": 5}
> wow.foo = None
> wow.foo
# {"wow": 1}
```

What do you think?